### ISSUE TITLE
SourcePlugin of MessageBusEventArgs defaults to calling assembly, not plugin

### ISSUE THREAD

**Issue by rappen** [MscrmTools/XrmToolBox#364]
The default behaviour seems to set the title of the **calling assembly** as ``SourcePlugin`` instead of the title of the **calling plugin**.
And if you change the ``SourcePlugin`` name to something else before calling the other plugin, FXB prevents executing with comment explanation "For some reason incorrect name was set in Source Plugin field".

When you have several XTB plugins in the same assembly, the value of the ``SourcePlugin`` property is not granular enough to identify which plugin that made the call.

Ping @shytikov

**Comment by shytikov** [comment]
yes, this should be fixed. I will look at this in my free time. 

And correct me if I'm working I guess unique Ids for plugins are also implemented, so we can do whole message bus thing more precise, right?

Ping @MscrmTools

**Comment by MscrmTools** [comment]
I think you are right but I worked on adding a unique Id long time ago

**Comment by MscrmTools** [comment]
@shytikov Did you find free time?
